lib/WebServer/ConfigWebServer.cpp

```diff
@@ -0,0 +1,240 @@
+
+  server.on("/system", HTTP_GET, authenticated([this]() {
+              server.send_P(200, TEXT_PLAIN, PSTR("POST your commands here"));
+            }));
+
+  server.on(
+      "/system", HTTP_POST,
+      authenticated(std::bind(&::ConfigWebServer::onSystemCommand, this)));
+
+  server.on("/config", HTTP_GET, authenticated([&]() {
+              String buff;
+              StringStream stream(buff);
+              settings.serialize(stream, true, false);
+              server.send(200, APPLICATION_JSON, buff);
+            }));
+
+  server.on("/config", HTTP_PUT, authenticated([&]() {
+              settings.deserialize(server.arg("plain"), true);
+              settings.save();
+              server.send(200, APPLICATION_JSON, F("true"));
+            }));
+
+  server.on("/protocols", HTTP_GET, authenticated([this]() {
+              const RfHandler* handler(getRfHandler());
+              if (handler) {
+                server.send(200, APPLICATION_JSON,
+                            handler->availableProtocols());
+              } else {
+                server.send(200, APPLICATION_JSON, F("[]"));
+              }
+            }));
+
+  server.on("/debug", HTTP_GET, authenticated([this]() {
+              const RfHandler* handler(getRfHandler());
+
+              if (!handler) {
+                server.send(200, APPLICATION_JSON, F("{}"));
+                return;
+              }
+
+              DynamicJsonBuffer buff;
+              JsonObject& root = buff.createObject();
+              root[F("protocolRaw")] = handler->isRawModeEnabled();
+
+              String result;
+              root.printTo(result);
+
+              server.send(200, APPLICATION_JSON, result);
+            }));
+
+  server.on("/debug", HTTP_PUT, authenticated([this]() {
+              RfHandler* handler(rfHandlerProvider());
+
+              if (!handler) {
+                server.send(500, APPLICATION_JSON, F("false"));
+                return;
+              }
+
+              DynamicJsonBuffer buff;
+              JsonObject& parsed = buff.parse(server.arg("plain"));
+
+              if (!parsed.success()) {
+                server.send(500, APPLICATION_JSON, "false");
+                return;
+              }
+
+              if (parsed.containsKey(F("protocolRaw"))) {
+                handler->setRawMode(parsed.get<bool>(F("protocolRaw")));
+              }
+
+              server.send(200, APPLICATION_JSON, F("true"));
+            }));
+
+  server.on("/firmware", HTTP_GET, authenticated([this]() {
+              server.send(200, APPLICATION_JSON, F("{\"version\": \"TODO\"}"));
+            }));
+
+  server.on(
+      "/firmware", HTTP_POST, authenticated([this]() {
+        server.sendHeader("Connection", "close");
+
+        Logger.info.println(F("Got an update. Reboot."));
+        if (Update.hasError()) {
+          server.send_P(
+              200, TEXT_PLAIN,
+              PSTR("Update failed. You may need to reflash the device."));
```

Comment:
Since we restart later, we should tell this here.

Reply:
This case is generally "please reflash" I would say. It doesnt matter if we reboot or not.

Reply:
This is a typical panic situation: The user wants to update via ota and it fails with: "Update failed. You may need to reflash the device." With this message is not clear if the device is now bricked and he need to reflash via cable. In addition, the box do a restart, thus the web interface does not work for a short time, leading the user to really think that he bricked the device.